codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddUserAgentDependency.java

```diff
@@ -33,12 +35,15 @@
 /**
  * Add client plubins and configs to support injecting user agent.
  */
+// TODO: Looks to add this back for non-AWS service clients, by fixing the dependency on ClientSharedValues.serviceId
```

Comment:
should be fairly easy to just write out the service shape name / id. This is what Go does